packages/hoppscotch-sh-admin/src/components/settings/ServerRestart.vue

```diff
@@ -79,22 +83,22 @@ onMounted(async () => {
 
   if (props.reset) {
     success = await resetInfraConfigs(resetInfraConfigsMutation);
-    if (!success) return;
+    if (!success) emit('mutationFailure');
   } else {
     const infraResult = await updateInfraConfigs(updateInfraConfigsMutation);
 
-    if (!infraResult) return;
+    if (!infraResult) emit('mutationFailure');
 
     const authResult = await updateAuthProvider(
       updateAllowedAuthProviderMutation
     );
-    if (!authResult) return;
+    if (!authResult) emit('mutationFailure');
 
     const dataSharingResult = await updateDataSharingConfigs(
       toggleDataSharingMutation
     );
 
-    if (!dataSharingResult) return;
+    if (!dataSharingResult) emit('mutationFailure');
```

Comment:
We'll have to return early after emitting the event. Currently, the control flow will proceed with the timer resulting in a page reload after `30s`.